bsip-0048.md

```diff
@@ -10,45 +10,76 @@
-of our platform. In the case of an ico, the ico-holders can create an asset with
-a fixed maximum supply. However this maximum supply can be changed by the
-asset creator (issuer) anytime. This exposes participators ico to the risk,
-of loosing value of their bought asset.
+The possibility of creating assets on the BitShares Blockchain is a core feature of our platform. However some things are too malleable when they should not be. Other things are not flexible when they should be.
 
-With this BSIP a flag will be introduced which forbids the modification of the maximum
-supply of an issuer created asset.
+Asset creators can create an asset with a maximum supply. However this maximum supply can be changed by the asset creator (issuer) anytime. This exposes token holders to the risk of their purchased asset losing value.
+
+With this BSIP a flag will be introduced which forbids the modification of the maximum supply of an issuer created asset.
+
+In addition, another new flag will be introduced that prevents any more tokens from being issued in the future.
+
+And finally, the ability to change the precision of an asset will now be available as long as current supply is zero.
 
 # Motivation
 
-The implementation of a flag, to disable the modification of the maximum supply for 
-issuer created asset, has effects for both sides. On the one side the security for the
-participators or non-issuer users increases, because the value of their assets (tokens)
-can't be manipulated by exposing the market to more created tokens. And on the other side
-it is good marketing for the ico-hoders, because they can advertise with their fixed
-maximum supply.
+A flag to disable the modification of the maximum supply for an asset will be implemented. This has benefits for both the participants and the asset creator. Asset holders are assured the value of their assets (tokens) will not be diluted by issuers creating more tokens. Asset creators are able to market the fact that the maximum supply is fixed.
+
+This can be very important in the case of Initial Coin Offerings (ICOs). Investors are assured by the network that their investment will not be diluted by new supply of that coin.
+
+A flag to disable issuing allows asset holders to be assured that the current supply will not increase, even if there are unissued tokens.
+
+The ability to change the precision of a created asset adds flexibility in the asset creation process.
 
 # Rational
 
-The `disable_modify_max_supply` is a new created flag in the `asset_issuer_permissions`.
-The flag can be activated to forbid the modification of the `max_supply` (`asset_object.options`).
-Once the flag is activated, it can't be deactivated.
+`disable_modify_max_supply` will be added to `asset_issuer_permissions`. The permission can be activated to forbid the modification of the `max_supply` (`asset_object.options`). The related flag can only be set to `false` if `current_supply` is zero.
```

Comment:
I think you have mixed up the meaning of permissions and flags.

This negative logic puts knots in my brain just from reading the BSIP. It will also be a PITA to implement. I think we should move these flags to positive logic. How many clients for creating assets are out there that might break?

Reply:
> How many clients for creating assets are out there that might break?

All clients.

Reply:
I think requiring clients to upgrade is better than being forever being stuck with this twisted logic.

Reply:
I think I clarified my permissions/flags statement. Please check. I have also switched to using positive logic. See https://github.com/bitshares/bsips/pull/216/commits/858b7ee5dc39a6b2acdb6d0ccfd8c5fd6083a0d3

Reply:
> I think it's better to revert to negative logic.

I agree here. If that twists user's minds we can present it positively in UI